org.eclipse.lsp4e/src/org/eclipse/lsp4e/operations/symbols/LSPSymbolInWorkspaceDialog.java

```diff
@@ -128,34 +129,31 @@ protected void fillContentProvider(AbstractContentProvider contentProvider, Item
 		if (itemsFilter.getPattern().isEmpty()) {
 			return;
 		}
-
-		for (LanguageServer server : this.languageServers) {
-			if (monitor.isCanceled()) {
-				return;
-			}
-
-			final var params = new WorkspaceSymbolParams(itemsFilter.getPattern());
-			try {
-				List<? extends WorkspaceSymbol> items = server.getWorkspaceService() //
-						.symbol(params) //
-						.thenApplyAsync(LSPSymbolInWorkspaceDialog::eitherToWorkspaceSymbols) //
-						.get(1, TimeUnit.SECONDS);
-				if(items != null) {
-					for (Object item : items) {
-						if (item != null) {
-							contentProvider.add(item, itemsFilter);
-						}
-					}
-				}
-			} catch (ExecutionException e) {
-				LanguageServerPlugin.logError(e);
-			} catch (InterruptedException e) {
-				LanguageServerPlugin.logError(e);
-				Thread.currentThread().interrupt();
-			} catch (TimeoutException e) {
-				LanguageServerPlugin.logWarning("Could not get workspace symbols due to timeout after 1 seconds in `workspace/symbol`", e); //$NON-NLS-1$
-			}
-		}
+		final var params = new WorkspaceSymbolParams(itemsFilter.getPattern());
+		this.executor.computeAll((w, ls) -> ls.getWorkspaceService()
+ 				.symbol(params)).stream().map(s -> s.thenApply(LSPSymbolInWorkspaceDialog::eitherToWorkspaceSymbols))
+ 			.forEach(cf -> {
+ 				if (monitor.isCanceled()) {
+ 					return;
+ 				}
+ 				try {
```

Comment:
since you talked about cancellation support, I have also been thinking about cancellation support and the get with timeout and its corresponding exception handling. 

I wonder if it would fit nicely addressed together. Now we have a rather arbitrary timeout and it the timeout is not enough, the answer to the request will never be read (but the request is not cancelled).
Shall it not be something like
```
while ("respose still useful") {
   try {
     get response with timeout
     do something with response
     exit while loop
   } catch (ExecutionException e) {
       LanguageServerPlugin.logError(e);
   } catch (InterruptedException e) {
      LanguageServerPlugin.logError(e);
      Thread.currentThread().interrupt();
   }  catch (TimeoutException) {
      if (!"respose still useful") {
         cancel request
         exit while loop
      }
     // else do nothing, retry
   }
}
```
The main point, is that the user should never see a timeout exception, either the answer is still useful and the we wait for it, or it is not, and in that case it is canceled. Of course the java code can be prettier than my sketched code :)

This is of course not for this PR, but for the one when we introduce cancellation support.

Reply:
Cool, I hope to have something ready to submit in draft later today (I keep getting sidetracked by other LSP4e issues that crop up!) then we can talk through the implications.